test/testdrive/avro-sources.td

```diff
@@ -226,6 +226,99 @@ a b
 1 2
 2 3
 
+# Test an Avro source without a Debezium envelope starting at specified partition offsets.
+
+$ kafka-create-topic topic=non-dbz-data-multi-partition partitions=2
+
+$ kafka-ingest format=avro topic=non-dbz-data-multi-partition schema=${non-dbz-schema} timestamp=1 partition=1
+{"a": 4, "b": 1}
+
+$ kafka-ingest format=avro topic=non-dbz-data-multi-partition schema=${non-dbz-schema} timestamp=1 partition=0
+{"a": 1, "b": 2}
+
+> CREATE MATERIALIZED SOURCE non_dbz_data_multi_partition
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-non-dbz-data-multi-partition-${testdrive.seed}'
+  WITH (start_offset=0)
```

Comment:
This case is a little ambiguous. The test could cover two behaviors: either a single start_offset is applied to all partitions, or the single start_offset is applied to the first partition and the second partition is defaulting to 0

The same test but with a start_offset >0 would disambiguate and make the single actual behavior clear